<element name="heightmap" required="0">
  <description>A heightmap based on a 2d grayscale image.</description>
  <element name="uri" type="string" default="__default__" required="1">
    <description>URI to a grayscale image file</description>
  </element>
  <element name="size" type="vector3" default="1 1 1" required="1">
    <description>The size of the heightmap in world units</description>
  </element>
  <element name="pos" type="vector3" default="0 0 0" required="0">
    <description>A position offset.</description>
  </element>

  <element name="texture" required="*">
    <description>The heightmap can contain multiple textures. The order of the texture matters. The first texture will appear at the lowest height, and the last texture at the highest hieght. Use blend to control the hieight thresholds and fade between textures.</description>
    <element name="size" type="double" default="10" required="1">
      <description>Size of the applied texture in meters.</description>
    </element>
    <element name="diffuse" type="string" default="__default__" required="1">
      <description>Diffuse texture image filename</description>
    </element>
    <element name="normal" type="string" default="__default__" required="1">
      <description>Normalmap texture image filename</description>
    </element>
  </element>
  <element name="blend" required="*">
    <description>The blend tag controls how two adjacent textures are mixed. The number of blend elements should equal one less than the number of textures.</description>
    <element name="min_height" type="double" default="0" required="1">
      <description>Min height of a blend layer</description>
    </element>
    <element name="fade_dist" type="double" default="0" required="1">
      <description>Distance over which the blend occurs</description>
    </element>
  </element>
  <element name="use_terrain_paging" type="bool" default="false" required="0">
    <description>Set if the rendering engine will use terrain paging</description>
  </element>
</element>
